parse-jsons-for-pm.js

```diff
@@ -46,122 +46,101 @@ const json = {
         // ex: ["this", "cant", "0", "be", "hard"]
-        let ar = array[i].split('.');
+        let ar = element.split('.');
 
-        let newString = '';
-        for(let j=0;j<ar.length;j++)
+        let newString = [];
+        for(const addr of ar)
         {
             // If its a number, add brackets, otherwise add a dot.
-            Number(ar[j]).toString() != "NaN" ? newString += `[${ar[j]}]` : newString += '.' + ar[j];
+            Number(addr).toString() != "NaN" ? newString.push(`[${addr}]`) : newString.push(`.${addr}`);
         }
 
         // add to the array of new Paths
-        v.push(newString.substring(1));
+        v.push(newString.join('').substring(1));
     }
     return v;
 }
 
 // for a collective test on attributes
 // exist(json)
-// prints assertion of pm test for key exist.
-const exist = (json) => { 
-    let list = fix_f(f(json));
-    for(let i=0;i<list.length;i++)
-    {
-        console.log(`pm.expect(obj.${list[i]}).to.exist;\n`);
-    }
-}
-
-// for separate tests
-// exist_test(json)
-// prints pm test for key exist
-const exist_test = (json) => { 
+// returns string of pm test line
+function exist(json) { 
+    let strArr = [];
     let list = fix_f(f(json));
     
     // loop over all the Paths
-    for(let i=0;i<list.length;i++)
+    for(const element of list)
     {
         // value from keyPath
-        let result = eval(`json.${list[i]}`);
+        let result = eval(`json.${element}`);
         
         // if result is null and not a child
-        if(result == null && `json.${list[i]}`.split('.').length == 2)
+        if(result == null && `json.${element}`.split('.').length == 2)
         {
             // fix a previous bug so its not printed "--(response.).to--"
-            console.log(`pm.test('${list[i]}', () => {\
-                \n\tpm.expect(response).to.have.property('${list[i]}');\n});`);
+            strArr.push(`pm.expect(response).to.have.property('${element}');`);
         }
         
         // if value is null, check if key exists only
         else if(result == null)
         {
             // split the path into pieces by .
-            let s = `${list[i]}`.split('.');
+            let s = `${element}`.split('.');
             // put all except the last back together
             let first = s.slice(0, s.length-1).join('.');
             let last = s[s.length-1];
 
             // make the test
-            console.log(`pm.test('${list[i]}', () => {\
-                \n\tpm.expect(response.${first}).to.have.property('${last}');\n});`);
+            strArr.push(`pm.expect(response.${first}).to.have.property('${last}');`);
         } 
         else {
             // make the test
-            console.log(`pm.test('${list[i]}', () => {\
-                \n\tpm.expect(response.${list[i]}).to.exist;\n});`);
+            strArr.push(`pm.expect(response.${element}).to.exist;`);
         }
-    }
+    } // out of for loop
+    return strArr;

```

Comment:
returning a `string` is more or array of strings is more managable than direct to console.log